### PR TITLE
fix(v2): incorrect dropdown options positioning on iOS Safari when the keyboard is open

### LINKED ISSUE #4718: [5] Dropdown options has incorrect positioning on iOS Safari when the keyboard is open
Only happens on iOS safari, might be due to keyboard shifting viewport. Will need to be fixed.

See related (unresolved) issue https://github.com/floating-ui/floating-ui/issues/805

Further reading on how we may get the correct position, or tell popperjs to update:
https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver

### PR DESCRIPTION
**Pull request by karrui**
> Builds on top of #4782 so the tests are fixed

## Problem

This PR replaces `react-popper` package usage with `@floating-ui/react-dom-interactions`. After experimentation, the layout is properly being updated on iOS Safari, which closes #4718 properly. The dev experience of using the package was also much better compared to react-popper. It is also the official successor of popperjs, which means a more updated package.

Closes #4718

## Solution

**Breaking Changes** 

- [x] No - this PR is backwards compatible  

**Features**:

- feat: use `@floating-ui/react-dom-interactions` instead of `react-popper` for Select components

**Bug Fixes**:

- Fixes bug where dropdown options were out of sync with the current viewport on iOS Safari.

## Before & After Screenshots
Should have no changes tbh.

## Tests

No changes to tests

## Deploy Notes

**New dependencies**:

- `@floating-ui/react-dom-interactions`: replaces `react-popper` for rendering Select component's dropdown options

### CODE REVIEW

**Comment by karrui** [comment]
~~DropdownField tests are failing, will investigate~~
Fixed in #4782

**Comment by karrui** [comment]
Before i forget tmr: long selected options are no longer truncated (will probably be cause by chromatic)

Oh, issue already created: #4724

File: frontend/src/components/Dropdown/components/SelectPopover/SelectPopover.tsx [hunk 1]
```diff
@@ -1,48 +1,51 @@
...
+    }),
+    [strategy, x, y],
+  )
+
+  // Allows
+  useLayoutEffect(() => {
+    if (isOpen && refs.reference.current && refs.floating.current) {
+      return autoUpdate(refs.reference.current, refs.floating.current, update)
+    }
+  }, [isOpen, update, refs.floating, refs.reference])
```
Comment (timotheeg) [inline frontend/src/components/Dropdown/components/SelectPopover/SelectPopover.tsx:48]: /justNoting: I don't have a suggestion here, but reading this file on its own, it's a little hard to keep track what all the refs are about 😰 :
* `reference`
* `wrapperRef`
* `mergedReferenceRefs`
* `refs.reference`
* `refs.floating`

Reading [the doc](https://floating-ui.com/docs/react-dom) helped a bit!

**Review by timotheeg — APPROVED** [review]
lgtm

**Pull request merged**